### DISCUSSION TITLE
OTEL instrumentation libraries

### DISCUSSION THREAD

**Discussion by sumitm-iiit** [open-telemetry/opentelemetry-java-instrumentation#3157] [Q&A]
Hi All,
I could see OTEL instrumentation libraries available for specific framework in case of java. I was wondering how can I use those libraries in my application? Example of one of the libraries - https://mvnrepository.com/artifact/io.opentelemetry.instrumentation/opentelemetry-auto-servlet
Any pointers in this direction would be helpful.

**Comment by trask** [comment]
hey @sumitm-iiit! I just sent a PR to add (very minimal) docs for the instrumentation libraries: #3176

some of them have README files, and some do not, in which case you'll need to look at the source for now, feel free to post any questions you may have about them

**Reply by sumitm-iiit** [reply]
@trask : Thanks for taking time out to answer my question. I can see your PR, but I am not able to see any documentation on how to include one of the instrumentation libraries in my application and start getting traces for the same. Does the doc to explaining the same already exists and I am not able to find it ?

**Reply by trask** [reply]
check out the links on the new page: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/standalone-library-instrumentation.md

the links take you to the different library instrumentation modules

some of them have README files, and some do not, in which case you'll need to look at the source for now, feel free to post any questions you may have about them

also, just want to confirm that you are unable to use the javaagent? the onboarding experience for that is much easier (and better documented currently)

**Reply by sumitm-iiit** [reply]
@trask : Thanks for your reply, will checkout the links and get back in case of any issues. 
No I did not face any issues running with otel java agent, I am just trying to explore how the instrumentation with specific libraries would work.
